tuf/api/metadata.py

```diff
@@ -1301,3 +1301,32 @@ def to_dict(self) -> Dict[str, Any]:
     def update(self, fileinfo: TargetFile) -> None:
         """Assigns passed target file info to meta dict."""
         self.targets[fileinfo.path] = fileinfo
+
+    def add_key(self, role: str, key: Key) -> None:
+        """Adds new signing key for delegated role 'role'.
+
+        Raises:
+            UnknownRoleError: If there are no 'delegations' in the Targets obj.
+            KeyError: If there is no role 'role' in the Delegations object.
+        """
+        if self.delegations is None:
+            raise exceptions.UnknownRoleError("Delegations doesn't exists!")
+        self.delegations.roles[role].keyids.add(key.keyid)
+        self.delegations.keys[key.keyid] = key
+
+    def remove_key(self, role: str, keyid: str) -> None:
+        """Removes key from delegated role 'role' and updates the delegations
+        key store.
+
+        Raises:
+            UnknownRoleError: If there are no 'delegations' in the Targets obj.
+            KeyError: If there is no role 'role' in the Delegations object.
+        """
```

Comment:
I don't think we need to talk about the Delegations object here (or the error message) at all: it's 99% an implementation detail.

There's two errors that caller might make: 
* role is not delegated by this Targets
* key is not actually used by role

Both maybe worth a mention in the docstring

Reply:
If the role is not delegated by this Target we will throw a KeyError. Will update the docstring.

If the key is not used by the given role, then there is nothing to be removed. I don't think we need to add any documentation for that.

Reply:
If the key is not used by the given role, we do raise something. ValueError probably?

Reply:
Yes, I am mistaken. We do throw a `KeyError` when trying to delete on line 1332.
Wondering should I make it an "if" check and throw a `ValueError` instead...